### ISSUE TITLE
Need for HTTP address for push notifications is incorrect.

### ISSUE THREAD

**Issue by sotatech** [home-assistant/home-assistant.github.io#28749]
### Feedback

I have the following cameras: RLC-842A, RLC-822A and RLC-1224A. My HA server ver  2023.08.4 uses a HTTPS address (with a certificate) and the motion detection (and all other sensors) works fine the three cameras.
The cameras are running the latest firmware.

### URL

https://www.home-assistant.io/integrations/reolink/

### Version

2023.8.4

**Comment by starkillerOG** [comment]
@sotatech that is because in your case the code automatically falls back to ONVIF long polling (PullPointSubscription).
This is also a form of push updates, but it is slightly slower than the ONVIF push (WS base notifications).
In my tests ONVIF push had a latency<0.3 seconds while ONVIF long polling was around 0.5 seconds.
Furthermore ONVIF push uses a single (re-)subscription message to the camera every 15 minutes and then the camera sends messages upon motion/ai detection on/off, while ONVIF long polling needs a active PullPointMessage beeing send to the camera every 20 seconds with aditional messages needed when motion/ai is changing faster than 20 seconds.

To make it more comples: When ONVIF long polling detects motion/AI, it has a cooldown time of 0.75 seconds before it can receive the next event. So if your camera detects motion, a person, a vehicle and a pet at the exactly the same time it can take up to 3x0.75 = 2.25 seconds extra to receive the last event (pet). For a NVR with up to 36 connected cameras this can add up when many cameras are detecting events rapidely after eachother.
ONVIF push does not have this limitation.

Bottom line conclusion: ONVIF push (ws base notifications) is slightly better than ONVIF long polling (PullPointSubscription).
That is why that section is there in the docs.

The above explanation is too technicall for in the docs, so I simplified it a bit there.

Hope this answers your questions.
If you want you can enable debug logging of one of the cameras and post the .txt log file here, than I can confirm for you what that camera is using (ONVIF push or ONVIF long polling).

If you appreciate the reolink integration and want to support its development, please consider [sponsering](https://github.com/sponsors/starkillerOG) the upstream library or purchase Reolink products through [this affiliate link](https://reolink.pxf.io/q44QWq).

**Comment by sotatech** [comment]
Thank you for that detailed explanation. The difference in performance does not seem very great and from watching the the sensor change in HA, it's response seems almost intantaneous.
